sdk/utils.ts

```diff
@@ -1,12 +1,38 @@
 import fetch, { RequestInit, Response } from 'node-fetch';
+import { FlagsmithTraitValue, ITraitConfig } from './types';
 // @ts-ignore
 if (typeof fetch.default !== 'undefined') fetch = fetch.default;
 
-export function generateIdentitiesData(identifier: string, traits: { [key: string]: any }) {
-    const traitsGenerated = Object.entries(traits).map(trait => ({
-        trait_key: trait[0],
-        trait_value: trait[1]
-    }));
+type Traits = { [key: string]: ITraitConfig | FlagsmithTraitValue };
+
+export function isTraitConfig(
+    traitValue: ITraitConfig | FlagsmithTraitValue
+): traitValue is ITraitConfig {
+    return !!traitValue && typeof traitValue == 'object' && traitValue.value !== undefined;
+}
+
+export function generateIdentitiesData(identifier: string, traits: Traits, transient: boolean) {
+    const traitsGenerated = Object.entries(traits).map(([key, value]) => {
+        if (isTraitConfig(value)) {
+            return {
+                trait_key: key,
+                trait_value: typeof value == 'object' ? value?.value : value,
+                transient: true
```

Comment:
This code is incorrect. 

```suggestion
                transient: value?.transient
```

Reply:
solved.